api/agent/call.go

```diff
@@ -322,6 +322,10 @@ func (c *call) Start(ctx context.Context, t callTrigger) error {
 		return fmt.Errorf("BeforeCall: %v", err)
 	}
 
+	if errInsert := c.ds.InsertCall(ctx, c.Model()); errInsert != nil {
+		common.Logger(ctx).WithError(err).Error("error inserting call during call.Start")
+	}
```

Comment:
to date, we've intentionally not done this because it adds latency for hot functions. our docs explicitly state we don't have any writes to db in the hot path for functions, and it's true that a user can get this behavior without this patch [if they use hot and reply in their container correctly] -- we are adding the call afterwards, but this can be after a client gets a response if done properly [by the user]. 

anyway, I think for async we should do this, but not sync. this patch does unconditionally